### PR TITLE
import: cache storage when possible

### LINKED ISSUE #13798: sst_importer: make shared external storage during restore
## Development Task
For now, during a restoration, even all files are stored at the same external storage, we still create external storage every time for each file.

It has been provided that creating external storage isn't a trivial operation. (exactly for remote external storages, which involved creating TLS connections -- we perhaps need to generate tokens before reading from object storages.) If we can reuse the external storage, we can reduce:
- The number of authorization request.
- The CPU cost.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close #13798

What's Changed:

This PR:

```commit-message
- Make the import process asynchronous.
- Added caching if client requires.
```

For making the download process be able to be executed asynchronously, we should add a `Send` trait for the returned `AsyncRead` of `read`. We have made two type alias `ExternalData` and `BlobStream` for flexibility.

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/sst_importer/src/caching/cache_map.rs [hunk 1]
```diff
@@ -0,0 +1,206 @@
+use std::{
```
Comment (3pointer) [inline components/sst_importer/src/caching/cache_map.rs:1]: plz add license

**Review by Leavrth — APPROVED** [review]

**Comment by 3pointer** [comment]
CI failed
``` 
thread 'sst_importer::tests::test_download_file_from_external_storage_for_sst' panicked at 'assertion failed: `(left != right)`

[2022-12-01T06:13:23.760Z]   left: `Plaintext`,

[2022-12-01T06:13:23.760Z]  right: `Plaintext`', components/sst_importer/src/sst_importer.rs:1346:13
```

**Comment by YuJuncen** [comment]
> CI failed
> 
> ```
> thread 'sst_importer::tests::test_download_file_from_external_storage_for_sst' panicked at 'assertion failed: `(left != right)`
> 
> [2022-12-01T06:13:23.760Z]   left: `Plaintext`,
> 
> [2022-12-01T06:13:23.760Z]  right: `Plaintext`', components/sst_importer/src/sst_importer.rs:1346:13
> ```

Fixed (BTW I'm wondering how this can pass CI yesterday...)

**Review by 3pointer — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**